init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/loader/Loader.java

```diff
@@ -18,30 +20,55 @@
     private final RoleServiceInvoker roleServiceInvoker;
     private final SiteServiceInvoker siteServiceInvoker;
     private final Trial trialConfig;
+    private final InitProgressService initProgressService;
 
     @Value("${delay-start:1}")
     private long delayStartInSeconds;
 
     @Autowired
     public Loader(Trial trialConfig, PractitionerServiceInvoker practitionerServiceInvoker,
-                  RoleServiceInvoker roleServiceInvoker, SiteServiceInvoker siteServiceInvoker) {
+                  RoleServiceInvoker roleServiceInvoker, SiteServiceInvoker siteServiceInvoker,
+                  InitProgressService initProgressService) {
         this.trialConfig = trialConfig;
         this.practitionerServiceInvoker = practitionerServiceInvoker;
         this.roleServiceInvoker = roleServiceInvoker;
         this.siteServiceInvoker = siteServiceInvoker;
+        this.initProgressService = initProgressService;
     }
 
     @Override
-    public void run(String... args) throws InterruptedException {
-        // TODO: remove this temporary fix in this story: https://ndph-arts.atlassian.net/browse/ARTS-362
+    public void run(String... args) throws InterruptedException, IOException {
         // Give the other services some time to come online.
+        initProgressService.submitProgress(String.format("init service delaying for %d seconds.", delayStartInSeconds));
         Thread.sleep(delayStartInSeconds * 1000);
 
-        roleServiceInvoker.execute(trialConfig.getRoles());
-        List<String> siteIds = siteServiceInvoker.execute(trialConfig.getSites());
-        // This yuk but it is the assumption in story https://ndph-arts.atlassian.net/browse/ARTS-164
-        String siteIdForUserRoles = siteIds.get(0);
-        practitionerServiceInvoker.execute(trialConfig.getPersons(), siteIdForUserRoles);
+        try {
+            initProgressService.submitProgress("getting roles from config.");
+            var roles = trialConfig.getRoles();
+
+            initProgressService.submitProgress("creating roles.");
+            roleServiceInvoker.execute(roles);
+
+            initProgressService.submitProgress("getting sites from config.");
+            var sites = trialConfig.getSites();
+
+            initProgressService.submitProgress("creating sites.");
+            List<String> siteIds = siteServiceInvoker.execute(sites);
+
+            initProgressService.submitProgress("getting persons from config.");
+            var persons = trialConfig.getPersons();
+
+            initProgressService.submitProgress("creating practitioner.");
+            // This yuk but it is the assumption in story https://ndph-arts.atlassian.net/browse/ARTS-164
```

Comment:
```suggestion
            // Assumption in story https://ndph-arts.atlassian.net/browse/ARTS-164
```

Reply:
written in a previous commit but someone else, I'll change it 😊